HSLdevcom/jore-map-ui
# 749 link validation

Closes #749

BE PR: https://gitlab.hsl.fi/jore/jore-map-backend/merge_requests/46

How to test:

Replace frontend's linkService method with:
```
    import _ from 'lodash';
    public static createLink = async (_link: ILink) => {
        const link = _.cloneDeep(_link);
        link.osNumber = '99999';
        console.log('at createLink ', link);
        await ApiClient.createObject(endpoints.LINK, link);
        await apolloClient.clearStore();
    }
```
--> try to create a new link
--> backend should prevent with "Bad Request"

Also go to linkView, replace existing method with:
```
    private transitTypeAlreadyExists = (transitType: TransitType) => {
        console.log(this.existingTransitTypes.includes(transitType));
        return false;
        // if (!this.props.isNewLink) return false;

        // return this.existingTransitTypes.includes(transitType);
    }
```
--> try to create a link that already exists
--> backend should prevent with "Bad Request"
